### ISSUE TITLE
Compatibility with new Fody

### ISSUE THREAD

**Issue by Delfistyaosani** [Fody/MethodDecorator#65]
I did break a few things but my fork works with newest Fody and net 4.7
Fixes issue #52 1 when build fails with 
Fody: The weaver assembly 'MethodDecorator.Fody, Version=0.9.1.6, Culture=neutral, PublicKeyToken=null' references an out of date version of Mono.Cecil.dll. Expected strong name token of '1C-A0-91-87-7D-12-CA-03' but got '50-CE-BF-1C-CE-B9-D0-5E'. The weaver needs to update to at least version 3.0 of FodyHelpers.
or
Fody: The weaver assembly 'MethodDecorator.Fody, Version=0.9.0.4, Culture=neutral, PublicKeyToken=null' references an out of date version of Mono.Cecil.dll 0.9.6.0. At least version 0.10 is expected. The weaver needs to add a NuGet reference to FodyCecil version 3.0.

Here's the fork: https://github.com/Delfistyaosani/MethodDecorator

**Comment by mpotta** [comment]
@Delfistyaosani  Why is the FodyWeavers.xml file not populated with the weaver tag <MethodDecorator /> in your fork ?

**Comment by Delfistyaosani** [comment]
@mpotta Because it's meant to be filled in actual project you compile with Fody. Usually it's like a switchboard with weavers being turned on and off by commenting/uncommenting. It looks like this in my test project:
<?xml version="1.0" encoding="utf-8" ?>
<Weavers>
  <MethodDecorator/>
  
</Weavers>

**Comment by mpotta** [comment]
Right, so my consumer project should ideally be auto-populated with the weaver on installing the NuGet, but that does not seem to be happening. 

Manually updating the XML causes a Error Fody: Could not find a weaver named 'MethodDecorator'.

**Comment by Delfistyaosani** [comment]
@mpotta Since this is not a complete update you need to do the following to make it work in consumer project (assuming you have built my fork by itself and DLLs ready):
1. Install official Fody from Nuget
2. Install official MethodDecorator from Nuget
3. Try to build and see post-build error "Method weaver fererences an out-of-date Mono bla bla"
4. Overwrite DLLs (MethodDecorator.Fody.dll, MethodDecorateorInterfaces.dll) from my fork info [test-project]\packages\MethodDecorator.Fody.0.9.0.6 folder 
5. Build test project (you must add FodyWeavers.xml to test project manually and populate with tags)

**Comment by Delfistyaosani** [comment]
Test project with compiled DLLs [https://drive.google.com/open?id=1AyPiGnQoZyqZBGb-Nqui05_aS-ICJVoI](url)

**Comment by MarcusNestinger** [comment]
@Delfistyaosani Your solution actually worked for me, thank you man!

**Comment by nagarjunan** [comment]
Hi Delfistyaosani,
I am also facing build issues. Could you please share the compiled DLLs because I am unable to open the above link.
